drivers/dai/nxp/sai/sai.c

```diff
@@ -142,6 +142,73 @@ static const struct dai_properties
 	CODE_UNREACHABLE;
 }
 
+#ifdef CONFIG_SAI_IMX93_ERRATA_051421
+/* notes:
+ *	1) TX and RX operate in the same mode: master/slave. As such,
+ *	there's no need to check the mode for both directions.
+ *
+ *	2) Only one of the directions can operate in SYNC mode at a
+ *	time.
+ *
+ *	3) What this piece of code does is it makes the SYNC direction
+ *	use the ASYNC direction's BCLK that comes from its input pad.
+ *	Logically speaking, this would look like:
+ *
+ *                      +--------+     +--------+
+ *                      |   TX   |     |   RX   |
+ *                      | module |     | module |
+ *                      +--------+     +--------+
+ *                         |   ^            |
+ *                         |   |            |
+ *                 TX_BCLK |   |____________| RX_BCLK
+ *                         |                |
+ *                         V                V
+ *                     +---------+    +---------+
+ *                     | TX BCLK |    | RX BCLK |
+ *                     |   pad   |    |   pad   |
+ *                     +---------+    +---------+
+ *                          |              |
+ *                          | TX_BCLK      | RX_BCLK
+ *                          V              V
+ *
+ *	Without BCI enabled, the TX module would use an RX_BCLK
+ *	that's divided instead of the one that's obtained from
+ *	bypassing the MCLK (i.e: TX_BCLK would have the value of
+ *	MCLK / ((RX_DIV + 1) * 2)). If BCI is 1, then TX_BCLK will
+ *	be the same as the RX_BCLK that's obtained from bypassing
+ *	the MCLK on RX's side.
+ *
+ *	4) The check for BCLK == MCLK is there to see if the ASYNC
+ *	direction will have the BYP bit toggled.
+ *
+ *	IMPORTANT1: in the above diagram and information, RX is SYNC
+ *	with TX. The same applies if RX is SYNC with TX. Also, this
+ *	applies to i.MX93. For other SoCs, things may be different
+ *	so use this information with caution.
+ *
+ *	IMPORTANT2: for this to work, you also need to enable the
+ *	pad's input path. For i.MX93, this can be achieved by setting
+ *	the pad's SION bit.
```

Comment:
As it is now, this works fine on i.MX93 I suppose, right?
The SION bit is set somewhere else? Where?

Reply:
Yep, works fine.
As for the SION bit: it's set via an IOMUXC register (e.g: `SW_MUX_CTL_PAD_GPIO_IO18`). We don't need to set it via this driver as it's already handled by Linux in our setup. One thing to watch out for here is the fact that the `cs42448` DTS for 93 doesn't set this bit so one would have to modify it to support the SYNC-ASYNC scenario (this is why the default configuration is ASNYC-ASYNC).